### ISSUE TITLE
[Bug] observed inaccurate bbox with offset in grounding task

### ISSUE THREAD

**Issue by lilyzhng** [OpenGVLab/InternVL#1084]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

![Image](https://github.com/user-attachments/assets/fbfa6dc7-4c46-47f6-9b63-d2add58a99bd)

When using opengvlab/internvl3-14b for 2d grounding task, I have observed inaccurate bbox with offset, is this expected? I am aware internvl uses 1000x1000 pixel coordinate system and performed below scaling

```
            abs_x1 = int(min(max(x1 / 1000 * width, 0), width))
            abs_y1 = int(min(max(y1 / 1000 * height, 0), height))
            abs_x2 = int(min(max(x2 / 1000 * width, 0), width))
            abs_y2 = int(min(max(y2 / 1000 * height, 0), height))
```

### Reproduction

**Model:** opengvlab/internvl3-14b
**Prompt:** "Outline the position of every vehicle in this image and output all the coordinates in JSON format with the structure: [{\"bbox_2d\": [x1, y1, x2, y2], \"label\": \"vehicle_type\"}]. Include cars, trucks, motorcycles, buses, and any other vehicles you can see."
**Response:** 
 To outline the position of every vehicle in the image, I'll identify each vehicle and provide their bounding boxes in the format [x1, y1, x2, y2], where (x1, y1) is the top-left corner and (x2, y2) is the bottom-right corner of the bounding box. Here's the JSON output:

```json
[
    {
        "bbox_2d": [10, 270, 200, 550],
        "label": "car"
    },
    {
        "bbox_2d": [230, 360, 360, 460],
        "label": "car"
    },
    {
        "bbox_2d": [290, 390, 350, 440],
        "label": "motorcycle"
    },
    {
        "bbox_2d": [330, 260, 920, 830],
        "label": "car"
    },
]


### Environment

```Shell
NA
```

### Error traceback

```Shell
NA
```

**Comment by muhammad-talha-ad** [comment]
Observed the same issue in InternVL3-1B and InternVL3-2B after finetuning on custom dataset well. 

- Unfreezing VIT layers in LORA finetuning did reduce the offset (but only slightly) 
- Reducing the temperature helps a lot in some cases but that also varies from image to image so not exactly an ideal solution.

Hoping for anyone else's input on this as well if they have faced and/or found a solution for this problem.

**Comment by muhammad-talha-ad** [comment]
Another observation that I forgot to mention is that reducing the number of maximum tiles made during dynamic preprocess (i.e. from default of 12 to let's say 6) also helps in significantly correcting the bbox offset issue on some images.

I think what might be happening is that when there are more tiles, the model might be getting confused in mapping of local coordinates of each tile back to the full image coordinates

Though I haven't tested this too thoroughly so I will share my findings once I do some more extensive testing

**Comment by curry2736** [comment]
> Observed the same issue in InternVL3-1B and InternVL3-2B after finetuning on custom dataset as well.
> 
> * Unfreezing VIT layers in LORA finetuning did reduce the offset (but only slightly)
> * Reducing the temperature helps a lot in some cases but that also varies from image to image so not exactly an ideal solution.
> 
> Hoping for anyone else's input on this as well if they have faced and/or found a solution for this problem.

@muhammad-talha-ad slightly unrelated -- you mentioned you did LoRA finetuning on the InternVL3 model. I don't believe the authors released a script for this. Did you modify the existing one/write your own? Did you have to use the `merge_lora.py` file for merging the weights after fine-tuning?

**Comment by curry2736** [comment]
@muhammad-talha-ad I've found that if you use `merge_lora.py` after LoRA finetuning, the model performance significantly degrades (98% accuracy -> 75% accuracy), at least for my task. My solution was to just use the unmerged LoRA finetuned model.

See https://github.com/huggingface/transformers/issues/31293 for more details
